### ISSUE TITLE
Image_Cached injector support

### ISSUE THREAD

**Issue by thats4shaw** [jonom/silverstripe-focuspoint#44]
I've inherited a project that uses this addon and after upgrading to SilverStripe 3.6.0 i've run into a bunch of interesting warnings.

[User Warning] DataObject::__construct passed The value 'random/image.jpg'. It's supposed to be passed an array, taken straight from the database. Perhaps you should use DataList::create()->First(); instead?

The error appears to be trigger by this change https://github.com/silverstripe/silverstripe-framework/pull/6845/commits/1d36f354e8349616c7b39fcade859fbcf0f9c362.

From my understanding this might make the FPImage_cached class in code/FPImage useful now?

To test this theory, if I uncomment the FPImage_cached class and swap out the class in yml as below my warnings disappear.

```
Injector:
  Image_Cached:
    class: FPImage_cached
```

Does all this sound sensible @jonom?

**Comment by jonom** [comment]
> From my understanding this might make the FPImage_cached class in code/FPImage useful now?

Yes, it does look like FPImage would be useful now!

However, I'm surprised that your fix would work. I wonder if it could be a coincidence? Sometimes with image stuff in SilverStripe, because the images are cached and there is some protection against re-trying crashing manipulations, you don't get the same result every time. For instance if the warning above is only generated when a manipulated image is being created and written to disk, you wouldn't see it on subsequent requests once it is already available and doesn't need to be generated.

Can you try flushing the cache while on the page that has the images that were causing an error, so the images are regenerated? If you can do this a few times with and without the fix in place it might tell us if it really works.

My gut feeling is that https://github.com/silverstripe/silverstripe-framework/commit/1d36f354e8349616c7b39fcade859fbcf0f9c362 may have introduced a regression. Image_cached is a bit of weird class that thinks it's a dataobject but isn't really, so maybe it trips up a bit when the Injector is used to create it.

**Comment by thats4shaw** [comment]
I flushed the cache multiple times yesterday before I delved into the rabbit hole yesterday however I just flushed again multiple times again to be sure with and without my fix and I can confirm the warnings still persist when my fix isn't in place.

I tend to agree with you about the introduction of a possible regression.

**Comment by minimalic** [comment]
The same error here after SilverStripe update from 3.5.3 to 3.5.4. Focuspoint 2.4.0 on SS 3.5.3 works well. So it seems the commit `1d36f35 ` has broken some image manipulation extensions like focuspoint and [https://github.com/restruct/silverstripe-svg-images/issues/2](https://github.com/restruct/silverstripe-svg-images/issues/2)

**Comment by jonom** [comment]
Hi guys, I just tried FocusPoint 2.4.0 with SS 3.5.4, SS3.6 and SS3.6-x-dev and couldn't replicate this issue anywhere.

Do you guys maybe have some other module installed that is messing with Image or Image_cached classes?

Can you try a clean install of SS with only this module added on and see if you're still getting errors?

**Comment by grgcnnr** [comment]
I'm experiencing the same thing. I'll see if i can figure out if any other modules are interfering / Do a minimal SS install.

Either way, Injector supports Image_cached so that's a win right?

:)

**Comment by grgcnnr** [comment]
Found it - for me at least.

If your your app.yml contains the override for Image like:
```
Injector:
  Image:
    class: FPImage
```

You'll get the error @thats4shaw @minimalic and myself are seeing.


Just tested on a barebones setup. based on SS3.5 can upload if you want.

**Comment by jonom** [comment]
Thanks @grgcnnr yep, that triggers the issue, and I was able to get @thats4shaw's fix working once I realised `_cached` has a capital C... doh! I will push a fix soon.

**Comment by jonom** [comment]
Fix is on the 2.5.0 release. Please note I renamed the commented-out `FPImage_cached` class to `FPImage_Cached` for consistency with `Image_Cached`.

You'll need:
```yml
Injector:
  Image:
    class: FPImage
  Image_Cached:
    class: FPImage_Cached
```
